.github/workflows/build-and-push-Docker-image.yml

```
@@ -0,0 +1,67 @@
+---
+name: Build and push docker image
+
+on:
+  workflow_call:
+    inputs:
+      image-name:
+        required: true
+        type: string
+      image-tag:
+        required: true
+        type: string
+    outputs:
+      full-image-name:
+        description: "Full pushed image name including host/registry, name, and tag"
+        value: ${{ jobs.docker.outputs.full-image-name }}
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    timeout-minutes: 25
+    outputs:
+      full-image-name: ${{ steps.image-name.outputs.full-image-name }}
+
+    steps:
+
+      - name: Define image name
+        id: image-name
+        run: |
+          full_image_name="ghcr.io/${{ github.repository_owner }}/${{ inputs.image-name }}:${{ inputs.image-tag }}"
+          full_image_name=$(echo $full_image_name | tr '[A-Z]' '[a-z]')
+          echo "full-image-name=$full_image_name" >> "$GITHUB_OUTPUT"
+          echo "$full_image_name" > full-image-name
+
+      - uses: actions/upload-artifact@v3
```

```suggestion
      - uses: actions/upload-artifact@v4
```
https://github.com/actions/upload-artifact/releases